components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,538 @@
+		overtime = Table.isNotEmpty(overtimes) and Json.stringify(overtimes) or nil
+	}
+	return map
+end
+
+function mapFunctions.getScoresAndWinner(map)
+	map.scores = {}
+	local indexedScores = {}
+	for scoreIndex = 1, MAX_NUM_OPPONENTS do
+		-- read scores
+		local score = map['score' .. scoreIndex]
+		local obj = {}
+		if not Logic.isEmpty(score) then
+			if TypeUtil.isNumeric(score) then
+				score = tonumber(score)
+				obj.status = _STATUS_HAS_SCORE
+				obj.score = score
+				obj.index = scoreIndex
+			elseif Table.includes(ALLOWED_STATUSES, score) then
+				obj.status = score
+				obj.score = -1
+				obj.index = scoreIndex
+			end
+			table.insert(map.scores, score)
+			indexedScores[scoreIndex] = obj
+		else
+			break
+		end
+	end
+	local isFinished = map.finished
+	if not Logic.isEmpty(isFinished) then
+		isFinished = Logic.readBool(isFinished)
+	else
+		isFinished = not Logic.readBool(map.unfinished)
+	end
+	if isFinished and not Logic.isEmpty(indexedScores) then
+		map.winner = mapFunctions.getWinner(indexedScores)
+	end
+
+	return map
+end
+
+function mapFunctions.getWinner(indexedScores)
+	table.sort(indexedScores, mapFunctions.mapWinnerSortFunction)
+	return indexedScores[1].index
+end
+
+function mapFunctions.mapWinnerSortFunction(op1, op2)
+	local op1norm = op1.status == _STATUS_HAS_SCORE
+	local op2norm = op2.status == _STATUS_HAS_SCORE
+	if op1norm then
+		if op2norm then
+			return tonumber(op1.score) > tonumber(op2.score)
+		else return true end
+	else
+		if op2norm then return false
+		elseif op1.status == _STATUS_DEFAULT_WIN then return true
+		elseif Table.includes(ALLOWED_STATUSES, op1.status) then return false
+		elseif op2.status == _STATUS_DEFAULT_WIN then return false
+		elseif Table.includes(ALLOWED_STATUSES, op2.status) then return true
+		else return true end
+	end
+end
+
+function mapFunctions.getTournamentVars(map)
+	map.mode = Logic.emptyOr(map.mode, Variables.varDefault('tournament_mode', '2v2'))
+	return MatchGroupInput.getCommonTournamentVars(map)
+end
+
+function mapFunctions.getParticipantsData(map)
+	local participants = map.participants or {}
+
+	for o = 1, MAX_NUM_OPPONENTS do
+		for player = 1, MAX_NUM_PLAYERS do
+			local participant = participants[o .. '_' .. player] or {}
+			local opstring = 'opponent' .. o .. '_p' .. player
+			if not Table.isEmpty(participant) then
+				participants[o .. '_' .. player] = participant
+			end
+		end
+	end
+
+	map.participants = participants
+	return map
+end
+
+--
+-- opponent related functions
+--
+function opponentFunctions.getTeamIcon(template)
+	local raw = mw.ext.TeamTemplate.raw(template)
+	if raw then
+		local icon = Logic.emptyOr(raw.image, raw.legacyimage)
+		local iconDark = Logic.emptyOr(raw.imagedark, raw.legacyimagedark)
+		return icon, iconDark
+	end
+end
+
+--the following 2 functions are a fallback
+--they are only useful if the team template doesn't exist
+--in the team template extension
+function opponentFunctions.getLegacyTeamName(template)
```

Comment:
this function and it's usage should imo be removed

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2259/commits/9614f42378bea81bfe2f83e33450f58c4cff63ed